source/javascripts/models/BitriseYmlService.ts

```diff
@@ -136,6 +136,32 @@ function addChainedWorkflow(
   return copy;
 }
 
+function updateStackAndMachine(workflowId: string, stack: string, machineTypeId: string, yml: BitriseYml): BitriseYml {
+  const copy = deepCloneSimpleObject(yml);
+
+  if (copy.workflows?.[workflowId]) {
+    const newBitriseIO: Meta['bitrise.io'] = {};
+    if (stack !== '') {
```

Comment:
I'd say, if the stack is truthy, we write it out, otherwise we don't.
```suggestion
    if (stack) {
```